### PR TITLE
[Model Runner V2] Simplify InputBuffers

### PR DESCRIPTION
**Pull request by WoosukKwon**
> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit e0473f60024cd01426b66de6d4336fb4b0199faa. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Modernizes buffer/copy strategy and simplifies input preparation across the GPU worker.
> 
> - Replace `UvaBackedGpuTensor` usage with plain GPU tensors; introduce `UvaBufferPool` and `copy_to_uva`/reworked `copy_to_gpu` for staging CPU→UVA→GPU copies
> - Rework `StagedWriteTensor` to use `UvaBufferPool` and apply writes directly (remove `prepare`); adjust reallocation logic for `write_contents`
> - Update `InputBuffers` to store direct tensors (`input_ids`, `positions`, `query_start_loc`, `seq_lens`) and compute `query_start_loc`/`seq_lens` on-device; remove UVA-backed fields
> - In `model_runner`, use temporary `UvaBufferPool` buffers for `idx_mapping`, `cu_num_logits`, and `query_start_loc`; propagate tensor-based `query_start_loc` to attention metadata and kernels
> - Switch multiple components from `.copy_to_gpu()` to `.copy_to_uva()` (e.g., `states.prefill_len`, sampling params) and from `.gpu` views to tensors (e.g., `spec_decode/eagle`, `cudagraph_utils`)
> - `block_table`: call `num_blocks.copy_to_uva()` after staged writes
> - `structured_outputs`: use `UvaBufferPool` for logits indices and grammar bitmasks, copying via UVA
> - Adjust kernels and call sites to accept tensors directly for `query_start_loc`, slot mappings, and indices
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e0473f60024cd01426b66de6d4336fb4b0199faa. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit 635e171e01d4138c6726163a11abecb4058f8c0b. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Modernizes GPU buffer management and input preparation with UVA-backed staging and direct tensor usage.
> 
> - Introduces `UvaBufferPool` with `copy_to_uva`/reworked `copy_to_gpu`; removes `UvaBackedGpuTensor` usages and `.gpu`/`.cpu` coupling in favor of plain tensors
> - Refactors `StagedWriteTensor` to use `UvaBufferPool` and apply writes directly; adjusts dynamic reallocation for `write_contents`
> - Simplifies `InputBuffers` to store `input_ids`, `positions`, `query_start_loc`, `seq_lens` as tensors and computes `query_start_loc`/`seq_lens` on-device
> - Updates `model_runner` to stage CPU→UVA→GPU copies via temporary pools for `idx_mapping`, `cu_num_logits`, `query_start_loc`; passes tensor `query_start_loc` through attention metadata and kernels
> - Switches several paths from `.copy_to_gpu()` to `.copy_to_uva()` in request `states` and sampling params; `block_table` now calls `num_blocks.copy_to_uva()` after writes
> - `cudagraph_utils`/`eagle`/`structured_outputs` updated to consume tensor/UVA buffers directly and accept tensor `query_start_loc`; grammar bitmask application now uses UVA pools
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 635e171e01d4138c6726163a11abecb4058f8c0b. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: vllm/v1/worker/gpu/cudagraph_utils.py [hunk 1]
```diff
@@ -228,10 +228,14 @@ def prepare_inputs_to_capture(
...
 ) -> dict[str, Any]:
     num_tokens_per_req = num_tokens // num_reqs
-    query_start_loc = input_buffers.query_start_loc
-    query_start_loc.np[: num_reqs + 1] = np.arange(num_reqs + 1) * num_tokens_per_req
-    query_start_loc.np[num_reqs:] = num_tokens
-    query_start_loc.copy_to_gpu()
+
+    query_start_loc_np = np.arange(num_reqs + 1) * num_tokens_per_req
+    query_start_loc_np[-1] = num_tokens
+    query_start_loc_cpu = torch.from_numpy(query_start_loc_np)
```
[the remark this answers is not in the document]
Comment (WoosukKwon) [reply vllm/v1/worker/gpu/cudagraph_utils.py:234]: Nice catch. Fixed.
